lib/portlayer/network/context.go

```diff
@@ -59,6 +61,9 @@ type Context struct {
 	scopes       map[string]*Scope
 	containers   map[string]*Container
 	defaultScope *Scope
+
+	portgroups              map[string]object.NetworkReference
+	DecodeNetworkReferences func(vctx context.Context, sess *session.Session, config *Configuration) map[string]object.NetworkReference
```

Comment:
This is really weird. I would have just replaced the global function in the unit tests.